--- a/common/lib/modules/fglrx/build_mod/firegl_public.c
+++ b/common/lib/modules/fglrx/build_mod/firegl_public.c
@@ -4498,8 +4498,8 @@
 
     if (cpu_has_pge)
     {
-        cr4 = read_cr4();
-        write_cr4(cr4 & ~X86_CR4_PGE);
+        cr4 = __read_cr4();
+        __write_cr4(cr4 & ~X86_CR4_PGE);
     }
      __flush_tlb();
 
@@ -4512,7 +4512,7 @@
     write_cr0(cr0 & 0xbfffffff);
     if (cpu_has_pge)
     {
-        write_cr4(cr4);
+        __write_cr4(cr4);
     }
     local_irq_restore(flags);
 
@@ -4539,8 +4539,8 @@
 
     if (cpu_has_pge)
     {
-        cr4 = read_cr4();
-        write_cr4(cr4 & ~X86_CR4_PGE);
+        cr4 = __read_cr4();
+        __write_cr4(cr4 & ~X86_CR4_PGE);
     }
      __flush_tlb();
   
@@ -4552,7 +4552,7 @@
     write_cr0(cr0 & 0xbfffffff);
     if (cpu_has_pge)
     {
-        write_cr4(cr4);
+        __write_cr4(cr4);
     }
     local_irq_restore(flags);
 
